### PR TITLE
[NETBEANS:4447]:Fixed NullPointerException while computing switch exp…

### PR DESCRIPTION
**Pull request by Mohan-Sarilla**
Fixed NullPointerException while computing switch expression conversion and formatting issue

### CODE REVIEW

**Review by Akshay-Gupta-Oracle — COMMENTED** [review]
Looks correct to me.
check the formatting, can be done automatically by Netbeans Formatter.

**Comment by Mohan-Sarilla** [comment]
> Looks correct to me.
> check the formatting, can be done automatically by Netbeans Formatter.

@Akshay-Gupta-Oracle I could see Formatting appearing fine in Netbeans.

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java:1907]: Please use not TABs but spaces.

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java:1910]: Please use not a TAB but spaces.

File: java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java
Comment (junichi11) [inline java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java:3140]: Please use not TABs but spaces.
Other changes as well.

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java:1906]: Please add `// NOI18N` if it is not expect i18N

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (Mohan-Sarilla) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java]: @junichi11 , done with the changes requested

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (Mohan-Sarilla) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java]: @junichi11 , done with the changes requested

File: java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java
Comment (Mohan-Sarilla) [reply java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java]: @junichi11 , done with the changes requested

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (Mohan-Sarilla) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java]: @junichi11 , done with the changes requested

File: java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java
Comment (junichi11) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java]: Thanks!

File: java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java
Comment (junichi11) [inline java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java:3160]: I would introduce a method for checking conditions because it's the same as above.
cc: @Akshay-Gupta-Oracle

**Comment by junichi11** [comment]
@Mohan-Sarilla Thank you for fixing it. Could you please squash your changes yourself?

**Comment by arvindaprameya** [comment]
Approved

**Comment by lahodaj** [comment]
It would be good to have tests (I think I can see what the change in Utilities is doing, but what exactly is doing the change in CasualDiff is less clear - and both are very easy to regress given there are no tests).

**Comment by lkishalmi** [comment]
@Mohan-Sarilla Can we have unit tests for this one?

**Comment by geertjanw** [comment]
@Mohan-Sarilla Can we have unit tests for this one?

**Comment by mishrasandeep** [comment]
> 
> 
> @Mohan-Sarilla Can we have unit tests for this one?

Have added the unit tests for the changes in the following PR: https://github.com/apache/netbeans/pull/2791

**Comment by geertjanw** [comment]
Is there any reason to not merge this?

**Comment by arvindaprameya** [comment]
Awaiting Jan's approval before going ahead and merging
________________________________
From: Geertjan Wielenga ***@***.***>
Sent: Wednesday, March 31, 2021 2:42 PM
To: apache/netbeans ***@***.***>
Cc: Arvind Aprameya ***@***.***>; Comment ***@***.***>
Subject: [External] : Re: [apache/netbeans] [NETBEANS:4447]:Fixed NullPointerException while computing switch exp… (#2570)


Is there any reason to not merge this?

**Review by singh-akhilesh — APPROVED** [review]
looks good.

**Pull request merged**
